tests/bofire/strategies/test_make_strategy.py

```diff
@@ -0,0 +1,118 @@
+import inspect
+import types
+import typing
+from typing import get_origin, get_type_hints
+
+import typing_extensions
+
+import bofire.data_models.strategies.api as dms
+from bofire.benchmarks.single import MultiTaskHimmelblau
+from bofire.data_models.features.task import TaskInput
+from bofire.strategies.api import (
+    AdditiveSoboStrategy,
+    DoEStrategy,
+    EntingStrategy,
+    MoboStrategy,
+    MultiFidelityStrategy,
+    MultiplicativeAdditiveSoboStrategy,
+    MultiplicativeSoboStrategy,
+    QparegoStrategy,
+    RandomStrategy,
+    SoboStrategy,
+)
+from bofire.strategies.fractional_factorial import FractionalFactorialStrategy
+from tests.bofire.strategies.test_base import domains
+
+
+def remove_optional(anno):
+    origin = get_origin(anno)
+    if origin == typing.Union or origin is types.UnionType:
+        return sorted((a for a in anno.__args__ if a is not type(None)), key=hash)
+    else:
+        return [anno]
+
+
+def test_make():
```

Comment:
I am wondering, could we use here for testing all **valid** strategy specs that are defined in https://github.com/experimental-design/bofire/blob/main/tests/bofire/data_models/specs/strategies.py ? As they are used in https://github.com/experimental-design/bofire/tree/main/tests/bofire/data_models/serialization and https://github.com/experimental-design/bofire/blob/main/tests/bofire/data_models/test_invalid.py. This would have the advantage, to not have to manually add new strategies also here, instead all strategy specs, that are registered for the serialization tests, would also be tested here. This would keep testing in the future easier.

Reply:
done.

Reply:
Btw., I by doing that I found some strategies that were missing. I will add them.